File: scsc-monolith/src/main/java/demo/scsc/api/productcatalog/ProductUpdateReceivedEvent.java
package demo.scsc.api.productcatalog;

import java.math.BigDecimal;
import java.util.UUID;

public record ProductUpdateReceivedEvent (
     UUID id,
     String name,
     String desc,
     BigDecimal price,
     String image,
     boolean onSale
){
}

File: scsc-monolith/src/main/kotlin/demo/scsc/api/productcatalog/ProductUpdateReceivedEvent.kt
package demo.scsc.api.productcatalog

import java.math.BigDecimal
import java.util.*

data class ProductUpdateReceivedEvent(
    val id: UUID,
    val name: String,
    val desc: String,
    val price: BigDecimal,
    val image: String,
    val onSale: Boolean
)
